test/test_foreach.py

```diff
@@ -0,0 +1,59 @@
+import torch
+import torch.cuda
+from torch.testing._internal.common_utils import TestCase, run_tests
+from torch.testing._internal.common_device_type import instantiate_device_type_tests, dtypes
+
+devices = (torch.device('cpu'), torch.device('cuda:0'))
+
+class TestForeach(TestCase):
+    @dtypes(*torch.testing.get_all_dtypes())
+    def test_add_scalar_same_size_tensors(self, device, dtype):
+        N = 20
+        H = 20
+        W = 20
+        tensors = []
+        for _ in range(N):
+            tensors.append(torch.zeros(H, W, device=device, dtype=dtype))
+
+        res = torch._foreach_add(tensors, 1)
+        for t in res:
+            if dtype == torch.bool and device == 'cpu':
+                dtype = torch.int64
```

Comment:
so this is another case where type promotion on cuda is different than on cpu - we should avoid that.